42 add H block with tests

Slawek-Sandbox opened the pull request:
Implemented H block (including tests)

TJB4epam commented on src/test/java/com/epam/prejap/tetris/block/HBlockTest.java:
```diff
@@ -0,0 +1,48 @@
+
+import static org.testng.Assert.*;
+
+@Test(groups = {"blockShapes"})
```
I'm a bit sick of those tests. Why don't we have a proper solution for once, with data providers, factories and one that ELIMINATES all the WhateverBlockTest classes and makes a parametrized test for them all?

Slawek-Sandbox replied:
I see @asilenko have already fixed that. I'm sorry - didn't have time to do it myself quicker.

TJB4epam reviewed (changes requested):
1. Please update the sources and resolve conflicts.
2. Where's the POM version bump?
3. See review remarks.

TJB4epam commented:
Conflicts reminder, please resolve.

Slawek-Sandbox commented:
I addressed all remarks
Ad 1 Conflicts resolved
Ad 2 POM version bump in
Ad 3 Didn't implement TestNG factory - I see Anna did, but I have refactored BlocksShapesTest class. I hope it counts ;)

TJB4epam reviewed (changes requested):
TY for the version.

Block, instead of String name and actual rows/cols is good for readability.
String.format - excellent choice!

Please, move the factory/dp combo that powers this class up - these are like a c-tor, so should be right after fields.

Actual tests should happen lower.

Overall - a good improvement on Anna's solution, that focuses on readability. I'd like to merge it fast.

Slawek-Sandbox commented:
TY for the review.

I moved factory/dp combo up, so test are at the bottom right now. I hope that's what you requested (I was not sure about private helper methods).

Slawek-Sandbox commented:
Summary:
- moved factory/dp combo up
- resolved conflicts (which appeared after [5-i-implement-i-block](https://github.com/EPAM-Online-Courses/tetris/pull/73) was merged)
- bumped POM version again

PR should be ready to merge.

TJB4epam reviewed (approved):
Thanks!
